tests/org.jboss.tools.openshift.test/src/org/jboss/tools/openshift/test/handler/PodLogsHandlerTest.java

```diff
@@ -0,0 +1,134 @@
+import com.openshift.restclient.model.IBuild;
+import com.openshift.restclient.model.IContainer;
+import com.openshift.restclient.model.IPod;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PodLogsHandlerTest {
+	
+	@Mock private IBuild build;
+	@Mock private IResourceWrapper<IBuild, ?> uiModel;
+	@Mock private IPod pod;
+	@Mock private IContainer container;
+	
+	private TestPodLogsHandler handler;
+	@SuppressWarnings("rawtypes")
+	private Map parameters = new HashMap();
+	private ExecutionEvent event;
+	
+	@Before
+	public void setUp() {
+		handler = spy(new TestPodLogsHandler());
+		event = new ExecutionEvent(null, parameters, null, null);
+	}
+
+	@Test
+	public void testGetLogsFromPodWhenRunning() throws ExecutionException {
+		givenAPodIsSelected("Running");
+		handler.execute(event);
+		thenTheLogsShouldBeShown();
+	}
+
+    @Test
+    public void testGetLogsFromPodWhenSucceeded() throws ExecutionException {
+        givenAPodIsSelected("Succeeded");
+        handler.execute(event);
+        thenTheLogsShouldBeShown();
+    }
+
+    @Test
+    public void testGetLogsFromPodWhenFailed() throws ExecutionException {
+        givenAPodIsSelected("Failed");
+        handler.execute(event);
+        thenTheLogsShouldBeShown();
+    }
+
+    @Test
+    public void testGetLogsFromPodWhenCompleted() throws ExecutionException {
+        givenAPodIsSelected("Completed");
+        handler.execute(event);
+        thenTheLogsShouldBeShown();
+    }
+
+    @Test
+    public void testGetLogsFromPodWhenPending() throws ExecutionException {
+        givenAPodIsSelected("Pending");
+        handler.execute(event);
+        thenInvalidStateShouldBeShown("Pending");
+    }
+
+    @Test
+    public void testGetLogsFromPodWhenUnknown() throws ExecutionException {
+        givenAPodIsSelected("Unknown");
+        handler.execute(event);
+        thenInvalidStateShouldBeShown("Unknown");
+    }
+
+    private void givenAPodIsSelected(String status) {
+	    doReturn(status).when(pod).getStatus();
+	    doReturn(Arrays.asList(container)).when(pod).getContainers();
+	    doReturn(pod).when(handler).getSelectedElement(any(ExecutionEvent.class), eq(IPod.class));
+	}
+
+	private void thenTheLogsShouldBeShown() {
+		verify(handler, times(1)).showLogs(any(), any());
+		verify(handler, times(0)).showDialog(any(ExecutionEvent.class), anyString(), anyString());
+	}
+
+    private void thenInvalidStateShouldBeShown(String status) {
+        verify(handler, times(1)).showLogs(any(), any());
+        verify(handler, times(1)).showDialog(any(ExecutionEvent.class), anyString(), eq(NLS.bind(PodLogsHandler.INVALID_POD_STATUS_MESSAGE,  status)));
+    }
+
+    public static class TestPodLogsHandler extends PodLogsHandler {
+
+        @Override
+        public <T> T getSelectedElement(ExecutionEvent event, Class<T> klass) {
+            return super.getSelectedElement(event, klass);
+        }
+
+        @Override
+        protected void showLogs(IPod pod, ExecutionEvent event) {
```

Comment:
@jeffmaury looks like sonar is right and we should be able to simply remove this method?